modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/editor/configuration/CKEditorConfigContributor.java

```diff
@@ -163,7 +173,8 @@ private JSONArray _getStyleFormatsJSONArray(Locale locale) {
 	}
 
 	private JSONArray _getToolbarSimpleJSONArray(
-		Map<String, Object> inputEditorTaglibAttributes) {
+		Map<String, Object> inputEditorTaglibAttributes,
+		boolean showAICreatorToolbar) {
```

Comment:
We can avoid passing custom flags if we apply the same pattern as `isShowSource()` method, that is, encapsulating the logic into a method:

```
boolean showAICreator(Map<String, Object> inputEditorTaglibAttributes) {
return GetterUtil.getBoolean(
			inputEditorTaglibAttributes.get(
				CKEditorConstants.ATTRIBUTE_NAMESPACE + ":showAICreator"));
}
```
Then using it from the places where we calculate the toolbars, as info is already available in `inputEditorTaglibAttributes` 

In addition, if  `isShowAICreator()` is moved to the the parent class, we'll give a chance to other editor config contributors inheriting from `BaseCKEditorConfigContributor` (not just the CKEditor one) to check and add that button to their respective toolbars. Not sure if you need this now, though.